src/main/java/org/apache/commons/text/similarity/SorensenDiceSimilarity.java

```diff
@@ -0,0 +1,146 @@
+     * {@link OverlapSimilarity}}
+     */
+    private final IntersectionSimilarity<Integer> similarity = new IntersectionSimilarity<>(this.converter);
+
+    /**
+     * Calculates Sorensen-Dice Similarity of two character sequences passed as
+     * input.
+     *
+     * <pre>
+     * similarity.apply(null, null)                 = IllegalArgumentException
+     * similarity.apply("foo", null)                = IllegalArgumentException
+     * similarity.apply(null, "foo")                = IllegalArgumentException
+     * similarity.apply("night", "nacht")           = 0.25
+     * similarity.apply("", "")                     = 1.0
+     * similarity.apply("foo", "foo")               = 1.0
+     * similarity.apply("foo", "foo ")              = 0.8
+     * similarity.apply("foo", "foo ")              = 0.66
+     * similarity.apply("foo", " foo ")             = 0.66
+     * similarity.apply("foo", " foo")              = 0.66
+     * similarity.apply("", "a")                    = 0.0
+     * similarity.apply("aaapppp", "")              = 0.0
+     * similarity.apply("frog", "fog")              = 0.4
+     * similarity.apply("fly", "ant")               = 0.0
+     * similarity.apply("elephant", "hippo")        = 0.0
+     * similarity.apply("hippo", "elephant")        = 0.0
+     * similarity.apply("hippo", "zzzzzzzz")        = 0.0
+     * similarity.apply("hello", "hallo")           = 0.5
+     * similarity.apply("ABC Corporation", "ABC Corp") = 0.7
+     * similarity.apply("D N H Enterprises Inc", "D &amp; H Enterprises, Inc.") = 0.74
+     * similarity.apply("My Gym Children's Fitness Center", "My Gym. Childrens Fitness") = 0.81
+     * similarity.apply("PENNSYLVANIA", "PENNCISYLVNIA") = 0.69
+     * </pre>
+     *
+     * @param left  the first CharSequence, must not be null
+     * @param right the second CharSequence, must not be null
+     * @return result similarity
+     * @throws IllegalArgumentException if either CharSequence input is {@code null}
+     */
+    @Override
+    public Double apply(final CharSequence left, final CharSequence right) {
+
+        if (left == null || right == null) {
+            throw new IllegalArgumentException("CharSequences must not be null");
+        }
+
+        if (StringUtils.equals(left, right)) {
+            return 1d;
+        }
+
+        // if bigram is not formed out of any given string, clearly both are not similar.
+        if (left.length() < 2 || right.length() < 2) {
+            return 0d;
+        }
+
+        IntersectionResult overlap = similarity.apply(left, right);
+
+        final int total = overlap.getSizeA() + overlap.getSizeB();
+        final long intersection = overlap.getIntersection();
+
+        return (2.0d * intersection) / total;
+    }
+
+    /**
+     * Converter class for creating Bigrams for SorensenDice similarity.
+     */
+    private static class SorensenDiceConverter implements Function<CharSequence, Collection<Integer>> {
```

Comment:
This can be a singleton. No need to instantiate it for each use.

`private static final SorensenDiceConverter INSTANCE = new SorensenDiceConverter()`

But this functionality could also be delivered by using a method reference. It is also not related to SorensenDice and should be in a package level utils class something like:

```
CharSequenceConverterUtils ...

List<Character> toCharacterList(CharSequence cs);
List<Integer> toBigramList(CharSequence cs);
List<String> toNGramList(CharSequence cs, int n);
```